src/http/server_context.rs

```diff
@@ -119,6 +61,92 @@ impl ServerContext {
       cache: ChronoCache::new(),
       grpc_data_loaders: Arc::new(grpc_data_loaders),
       env_vars: Arc::new(env),
+    };
+
+    for def in bp.definitions.iter_mut() {
+      if let Definition::ObjectTypeDefinition(def) = def {
+        for field in &mut def.fields {
+          if let Some(expr) = &field.resolver {
+            field.resolver = ctx.check_resolver_of_field(field, expr);
+          }
+        }
+      }
+    }
+    ctx.schema = bp.to_schema();
+    ctx.blueprint = bp;
+    ctx
+  }
+
+  fn check_resolver_of_field(&mut self, field: &blueprint::FieldDefinition, expr: &Expression) -> Option<Expression> {
```

Comment:
I think it could accept `&mut FieldDefinition` and the field.resolver could be just updated inside the function instead of returning new expr

Reply:
I can't agree with that because this method is already using mutation with `&mut self` and you can't call this immutable approach. 

I'd like to use consistent approach. If you still thinks current fn is better feel free to leave it as is

Reply:
Actually, I found an error in current code:

in `check_unsafe_expr` the last wildcard match arm returns None which will remove the existing resolver on this line `field.resolver = ctx.check_resolver_of_field(field, expr)`

That's easy to make such kind of mistakes when rewriting from mutable to immutable version of code =)